### PR TITLE
Fixing a query generation for the epn-tap data model.

### PR DESCRIPTION
**Pull request by msdemlei**
This closes bug #393

### CODE REVIEW

**Review by rsav — APPROVED** [review]
PR tested and works from me: 
`registry.search(registry.Datamodel('epntap')) `
now retrieves 244 rows.
Thanks msdemlei !

File: pyvo/registry/tests/test_regtap.py [hunk 1]
```diff
@@ -630,8 +646,12 @@ def test_ambiguous_access_url_warns(self, recwarn):
...
         })
         assert rsc.access_url == "http://a"
-        assert [str(w.message)[:50] for w in recwarn
-                ] == ['The resource ivo://pyvo/test_regtap.py has multipl']
+        # the following test is written this oddly to avoid that extra
+        # warnings break it.
+        expectation = set(
+            ['The resource ivo://pyvo/test_regtap.py has multipl'])
+        assert set(str(w.message)[:50] for w in recwarn
+            ) & expectation == expectation
```
Comment (andamian) [inline pyvo/registry/tests/test_regtap.py:654]: Isn't this a simpler way to check for the expected warning in the set?

```suggestion
        assert 'The resource ivo://pyvo/test_regtap.py has multipl' in [str(w.message)[:50] for w in recwarn]
```

Comment (bsipocz) [reply pyvo/registry/tests/test_regtap.py:654]: This feels a bit over engineered, any reasons not to use the `pytest.warns` context in tests?

**Comment by msdemlei** [comment]
On Tue, Jan 10, 2023 at 10:49:43AM -0800, Adrian wrote:
> ```suggestion
>         assert 'The resource ivo://pyvo/test_regtap.py has multipl' in [str(w.message)[:50] for w in recwarn]
> ```

Fair enough.  Force-pushed the change.

Thanks!

**Review by andamian — APPROVED** [review]

**Review by andamian — APPROVED** [review]
It looks good except for the style issues.

**Comment by bsipocz** [comment]
Given this is a user reported bug and also present in the latest release, it should have a changelog entry.

**Comment by msdemlei** [comment]
On Wed, Jan 11, 2023 at 03:43:09PM -0800, Brigitta Sipőcz wrote:
> Given this is a user reported bug and also present in the latest
> release, it should have a changelog entry.

Added one, force-pushed.  @andamian: is that the style issue you were
mentioning?  If so, would you merge?  Thanks!

**Comment by andamian** [comment]
> 

The style check action is still not green in CI. Check https://github.com/astropy/pyvo/actions/runs/3900161110/jobs/6660548272

**Comment by msdemlei** [comment]
On Thu, Jan 12, 2023 at 01:45:43AM -0800, Adrian wrote:
> The style check action is still not green in CI. Check https://github.com/astropy/pyvo/actions/runs/3900161110/jobs/6660548272

Ah dang, missed that.  I'll need to fix my commit routine...

**Comment by bsipocz** [comment]
Given this is all approved, I'll just go ahead and fix the minor issue and changelog conflict and merge it.

**Comment by msdemlei** [comment]
On Tue, Jan 24, 2023 at 11:22:40AM -0800, Brigitta Sipőcz wrote:
> Given this is all approved, I'll just go ahead and fix the minor issue and changelog conflict and merge it.

Thanks! (And sorry for having missed the problem).

**Pull request merged**